File: Blind75/ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Tip: First consider length
        if len(s) != len(t):
            return False
        
        set_s, set_t = {}, {}

        for char in s:
            if char in set_s:
                set_s[char] += 1
            else:
                set_s[char] = 1
            
        for char in t:
            if char in set_t:
                set_t[char] += 1
            else:
                set_t[char] = 1

        if set_s == set_t:
            return True
        else:
            return False

File: Blind75/ValidAnagram.cpp
using namespace std;
#include <string>
#include <map>

class Solution {
    public:
        bool isAnagram(string s, string t) {
            map<char, int> mp1 = {};
            map<char, int> mp2 = {};
    
            for (char c: s) {
                if (mp1.find(c) != mp1.end()) {
                    mp1[c]++;
                } else {
                    mp1.insert({c, 1});
                }
            }
    
            for (char c: t) {
                if (mp2.find(c) != mp2.end()) {
                    mp2[c]++;
                } else {
                    mp2.insert({c, 1});
                }
            }
    
            if (mp1 == mp2) {
                return true;
            }
    
            return false;
        }
    };